### ISSUE TITLE
ProgressBar() does not work with L_BFGS

### ISSUE THREAD

**Issue by tonideleo** [mlpack/ensmallen#361]
#### Issue description

Following the example.cpp, I was playing around with some callbacks functions. For example, I realized if I use the Adam optimizer, the callback function ProgressBar() works without issues; however, it does not work with L_BFGS. In the documentation I did not find any exception use for this callback nor limitation in the section regarding the L_BFGS.

#### Your environment

 * version of ensmallen: 2.19.0
 * operating system: Windows 10
 * compiler: MS Visual Studio 2022 (ensmallen installed via vcpkg and cmake)
 * version of Armadillo: 11.4.4
 * any other environment information you think is relevant:

#### Steps to reproduce
In line 64 of the example, substitute the following:
opt.Optimize(lrf, theta, ens::ProgressBar());

#### Expected behavior

I was expecting to see the progress bar, which by the way it worked if I used a different optimizer.

#### Actual behavior

![image](https://user-images.githubusercontent.com/55999079/221339130-b62a7209-8718-478d-9c6e-ce9296ad719a.png)

**Comment by conradsnicta** [comment]
@zoq may have more insight here.  It's possible that ens::ProgressBar() doesn't support L_BFGS

**Comment by zoq** [comment]
That is correct, `ens::ProgressBar()` doesn't support `L_BFGS`, but I don't see a reason why it couldn't. I'm opening a PR to add support.
